src/VisualStudio/Xaml/Impl/Features/InlineRename/XamlEditorInlineRenameService.cs

```diff
@@ -0,0 +1,186 @@
+            {
+                return true;
+            }
+
+            private static Glyph FromSymbolKind(SymbolKind kind)
+            {
+                var glyph = Glyph.Error;
+
+                switch (kind)
+                {
+                    case SymbolKind.Namespace:
+                        glyph = Glyph.Namespace;
+                        break;
+                    case SymbolKind.NamedType:
+                        glyph = Glyph.ClassPublic;
+                        break;
+                    case SymbolKind.Property:
+                        glyph = Glyph.PropertyPublic;
+                        break;
+                    case SymbolKind.Event:
+                        glyph = Glyph.EventPublic;
+                        break;
+                }
+
+                return glyph;
+            }
+
+            private class InlineRenameLocationSet : IInlineRenameLocationSet
+            {
+                private readonly Solution _oldSolution;
+
+                public InlineRenameLocationSet(Solution solution, IList<InlineRenameLocation> locations)
+                {
+                    _oldSolution = solution;
+                    Locations = locations;
+                }
+
+                public IList<InlineRenameLocation> Locations { get; }
+
+                public async Task<IInlineRenameReplacementInfo> GetReplacementsAsync(string replacementText, OptionSet optionSet, CancellationToken cancellationToken)
+                {
+                    var newSolution = _oldSolution;
+                    foreach (var group in Locations.GroupBy(l => l.Document))
+                    {
+                        var document = group.Key;
+                        var oldSource = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                        var newSource = oldSource.WithChanges(group.Select(l => new TextChange(l.TextSpan, replacementText)));
+                        newSolution = newSolution.WithDocumentText(document.Id, newSource);
+                    }
+
+                    return new InlineRenameReplacementInfo(this, newSolution);
+                }
+
+                private class InlineRenameReplacementInfo : IInlineRenameReplacementInfo
+                {
+                    private readonly InlineRenameLocationSet _inlineRenameLocationSet;
+
+                    public InlineRenameReplacementInfo(InlineRenameLocationSet inlineRenameLocationSet, Solution newSolution)
+                    {
+                        NewSolution = newSolution;
+                        _inlineRenameLocationSet = inlineRenameLocationSet;
+                    }
+
+                    public Solution NewSolution { get; }
+
+                    public IEnumerable<DocumentId> DocumentIds => _inlineRenameLocationSet.Locations.Select(l => l.Document.Id).Distinct();
+
+                    // TODO: actually check to make sure it is a legal identifier.
```

Comment:
I will move the responsibility to determine this out to the XAML language service in VS.